### PR TITLE
Load init fix issue 9 30

### PR DESCRIPTION
**Pull request by GitPascalP**
Fixed the initialization for the ExternalSpeedLoad and ConstantSpeedLoad.
Init in the environment is possible as a class instance (from requested load type) with the initializer as an argument, but Init should be also possible as keyword d for the ConstantSpeedLoad class with the load_intializer as an environment argument.
ExternalSpeedLoad is better supported by the initialization with an instance, due to the additional parameters for the speed profile.

### CODE REVIEW

File: gym_electric_motor/physical_systems/mechanical_loads.py [hunk 1]
```diff
@@ -11,10 +11,10 @@ class MechanicalLoad:
     It contains an mechanical ode system as well as the state names, limits and nominal values
     of the mechanical quantities. The only required state is 'omega' as the rotational speed of the motor shaft
     in rad/s.
-    ExternalSpeedLoad and ConstantSpeedLoad should be initialized with the own
-    initial value in class initialisation, when definded external
+    ExternalSpeedLoad and ConstantSpeedLoad can be initialized by giving the
+    initializer as an class parameter by instanziation
```
Comment (max-schenke) [inline gym_electric_motor/physical_systems/mechanical_loads.py:15]: As stated in the further annotations, externalSpeedLoad should be changed such that it ignores the initializer. 
(also please correct typo instanziation -> instantiation)

File: examples/external_speed_profile.py [hunk 2]
```diff
@@ -16,45 +17,58 @@
...
+# usage of a random load initializer is only recommended for the
+# ConstantSpeedLoad, due to the already given profile by an ExternalSpeedLoad
+load_init = {'random_init': 'uniform'},
+
 # external speed profiles can be given by an ExternalSpeedLoad, ConstantSpeedLoad
-# initial values are defined in the class
-ext_load = ExternalSpeedLoad(omega_initial=5, speed_profile=sinus_lambda,
-                             amp=30, f=2, b=30)
+# the initial value of the external load should be match with the profile
+# offset to avoid complicated spikes after a new episode start
```
Comment (max-schenke) [inline examples/external_speed_profile.py:33]: The hints given in these comments do not match the intended behavior of the externalSpeedLoad. Information on that are noted my other annotations.

**Review by max-schenke — CHANGES_REQUESTED** [review]
Behavior of constantSpeedLoad seems to be as intended. Behavior of externalSpeedLoad would need changes as no initializer is to be used there.

File: gym_electric_motor/physical_systems/mechanical_loads.py [hunk 3]
```diff
@@ -357,10 +359,15 @@ def __init__(self, speed_profile=None, omega_initial=0,
...
 
             omega_initial(float)): Initial value for the speed in rad/s.
+                the initial value must be the same as the speed profile
+                offset for omptimal usage
             tau(float): discrete time step of the system
             kwargs(dict): further arguments for speed_profile
         """
-        super().__init__()
+        self._initializer = self._default_initializer.copy()
+        self._initializer['states']['omega'] = omega_initial
```
[the remark this answers is not in the document]
Comment (GitPascalP) [reply gym_electric_motor/physical_systems/mechanical_loads.py:368]: usage of an initializer is deleted. Speed profile at time 0 is used as initial value

**Review by max-schenke — CHANGES_REQUESTED** [review]
Minor changes still required to improve user feedback. Functionality seems to work as intended, thanks for your contribution!

File: gym_electric_motor/physical_systems/mechanical_loads.py [hunk 4]
```diff
@@ -355,15 +354,20 @@ def __init__(self, speed_profile=None, omega_initial=0,
...
             tau(float): discrete time step of the system
             kwargs(dict): further arguments for speed_profile
         """
-        super().__init__()
+
+        super().__init__(**kwargs)
+        warnings.warn('Given initializer will be overwritten with starting value from speed-profile, '
+                      'to avoid complications at the load reset. It is recommended to choose starting value of'
+                      ' load by the defined speed-profile.',
+                      UserWarning)
```
[the remark this answers is not in the document]
Comment (GitPascalP) [reply gym_electric_motor/physical_systems/mechanical_loads.py]: changed as requested

File: gym_electric_motor/physical_systems/mechanical_loads.py [hunk 5]
```diff
@@ -341,12 +345,7 @@ def omega(self):
         """
         return self._omega_initial
 
-    def reset(self, *_, **__):
-        # Docstring from superclass
-        return np.array([self._omega_initial])
-
-    def __init__(self, speed_profile=None, omega_initial=0,
-                 tau=1e-4, **kwargs):
```
[the remark this answers is not in the document]
Comment (GitPascalP) [reply gym_electric_motor/physical_systems/mechanical_loads.py:349]: to minimize error and the probability to have different tau's initialized it makes sense to just do it one time. I thought the functionality of the instantiate function in physcial_system.py targeted this case. The load is instantiated with tau and other kwargs, but the external speed load doesn't take the tau, given by the physical system and I'm not quite sure why.

**Review by max-schenke — APPROVED** [review]
Thanks to @GitPascalP for his fast reaction times. The necessary features should be working now.

**Pull request merged**